File: JavaScript/Unique3-DigitEvenNumbers.js
const totalNumbers = function (digits) {
  const numbers = new Set();

  const n = digits.length;

  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {
      if (j === i) continue;
      for (let k = 0; k < n; k++) {
        if (k === i || k === j) continue;
        if (digits[i] === 0) continue;

        const number = digits[i] * 100 + digits[j] * 10 + digits[k];

        if (number % 2 === 0) numbers.add(number);
      }
    }
  }
  return numbers.size;
};
console.log(totalNumbers([1, 2, 3, 4]));

File: Python/Unique3-DigitEvenNumbers.py
class Solution:
    def totalNumbers(self, digits: list[int]) -> int:

        numbers = set()

        for i in range(len(digits)):
            if digits[i] == 0:
                continue
            for j in range(len(digits)):
                if i == j:
                    continue
                for k in range(len(digits)):
                    if k == i or k == j:
                        continue
                    number = digits[i] * 100 + digits[j] * 10 + digits[k]

                    if number % 2 == 0:
                        numbers.add(number)
        return len(numbers)


solution = Solution()

print(solution.totalNumbers([1, 2, 3, 4]))
